apis/tempo/v1alpha1/microservices_webhook.go

```diff
@@ -98,6 +100,24 @@ func (d *Defaulter) Default(ctx context.Context, obj runtime.Object) error {
 		r.Spec.SearchSpec.DefaultResultLimit = &defaultDefaultResultLimit
 	}
 
+	// Default replicas for distibutor
+	if r.Spec.Components.Distributor == nil {
+		r.Spec.Components.Distributor = &TempoComponentSpec{}
+	}
+
+	if r.Spec.Components.Distributor.Replicas == nil {
+		r.Spec.Components.Distributor.Replicas = pointer.Int32(1)
+	}
+
+	// Default replicas for ingester
+	if r.Spec.Components.Ingester == nil {
+		r.Spec.Components.Ingester = &TempoComponentSpec{}
+	}
+
+	if r.Spec.Components.Ingester.Replicas == nil {
+		r.Spec.Components.Ingester.Replicas = pointer.Int32(1)
+	}
+
```

Comment:
Should we also set `r.Spec.ReplicationFactor` to the default value of `1` here?

Reply:
oh yeah! good catch